addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairBaseBridgeHandler.java

```diff
@@ -0,0 +1,468 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+/**
+ /* Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.pentair.handler;
+
+import static org.openhab.binding.pentair.PentairBindingConstants.INTELLIFLO_THING_TYPE;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.pentair.internal.PentairPacket;
+import org.openhab.binding.pentair.internal.PentairPacket_Intellichlor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract class for all common functions for different bridge implementations. Use as superclass for IPBridge and
+ * SerialBridge implementations.
+ * <ul>
+ * <li>Implements parsing of packets on Pentair bus and dispositions to appropriate Thing
+ * <li>Periodically sends query to any {@link PentairIntelliFloHandler} things
+ * <li>Provides function to write packets
+ * </ul>
+ *
+ * Format protocol leveraged from the following: https://github.com/tagyoureit/nodejs-Pentair/blob/master/index.js
```

Comment:
This was an old comment, i reviewed the original source and can't identify anything that was reused.